metadoc-js/src/main/scala/monaco/Monaco.scala

```diff
@@ -2314,6 +2306,15 @@ package languages {
   }
 
   @js.native
+  trait TypeDefinitionProvider extends js.Object {
+    def provideTypeDefinition(
+        model: editor.IReadOnlyModel,
+        position: Position,
+        token: CancellationToken
+    ): Definition | Thenable[Definition] = js.native
```

Comment:
It seems this was different than I initially thought. "Go to type" seems to be a way to jump to the type definition when the cursor is lying above a term/pattern (non-type symbol). Example,

```scala
val x = List(1)
```

"Go to type" on the variable `x` will jump to the `List` symbol. Context https://github.com/Microsoft/TypeScript/issues/10394

I was hoping it was a global "Jump to type" search, but it's pretty useful nevertheless.